### PR TITLE
Go Modules update using go get -u; go mod tidy

### PR DESCRIPTION
**Pull request by grastogi23**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,6 +3,28 @@ module github.com/vmware/terraform-provider-avi
...
+	github.com/hashicorp/go-uuid v1.0.2 // indirect
+	github.com/hashicorp/hcl/v2 v2.10.1 // indirect
+	github.com/hashicorp/terraform-plugin-sdk/v2 v2.8.0
+	github.com/hashicorp/yamux v0.0.0-20210826001029-26ff87cf9493 // indirect
+	github.com/mattn/go-colorable v0.1.11 // indirect
+	github.com/mitchellh/go-testing-interface v1.14.1 // indirect
+	github.com/mitchellh/go-wordwrap v1.0.1 // indirect
+	github.com/mitchellh/mapstructure v1.4.2 // indirect
+	github.com/oklog/run v1.1.0 // indirect
+	github.com/vmware/alb-sdk v0.0.0-20211021091749-5269c93f23b0
```
Comment (mayank-avinetworks) [inline go.mod:22]: @yograjshisode , I believe we'd need to update and tidy the go.mod/sum before the provider release to update the latest commits say e.g. alb-sdk or other fixes in indirect dependencies. 
Are you OK to consume this change now or would want to pull in changes in a single commit at the time of release by changing them in avi-dev.

**Comment by yograjshisode** [comment]
run jenkins-terraform-test

**Comment by mayank-avinetworks** [comment]
seems like testing interface has been updated for TF.
```
10:52:37 # github.com/vmware/terraform-provider-avi/avi [github.com/vmware/terraform-provider-avi/avi.test]
10:52:37 avi/data_source_avi_actiongroupconfig_test.go:13:15: cannot use t (type *"testing".T) as type "github.com/mitchellh/go-testing-interface".T in argument to resource.Test:
10:52:37 	*"testing".T does not implement "github.com/mitchellh/go-testing-interface".T (missing Cleanup method)
10:52:37 avi/data_source_avi_alertconfig_test.go:13:15: cannot use t (type *"testing".T) as type "github.com/mitchellh/go-testing-interface".T in argument to resource.Test:
10:52:37 	*"testing".T does not implement "github.com/mitchellh/go-testing-interface".T (missing Cleanup method)
10:52:37 avi/data_source_avi_alertemailconfig_test.go:13:15: cannot use t (type *"testing".T) as type "github.com/mitchellh/go-testing-interface".T in argument to resource.Test:
10:52:37 	*"testing".T does not implement "github.com/mitchellh/go-testing-interface".T (missing Cleanup method)
10:52:37 avi/data_source_avi_alertscriptconfig_test.go:13:15: cannot use t (type *"testing".T) as type "github.com/mitchellh/go-testing-interface".T in argument to resource.Test:
10:52:37 	*"testing".T does not implement "github.com/mitchellh/go-testing-interface".T (missing Cleanup method)
10:52:37 avi/data_source_avi_analyticsprofile_test.go:13:15: cannot use t (type *"testing".T) as type "github.com/mitchellh/go-testing-interface".T in argument to resource.Test:
10:52:37 	*"testing".T does not implement "github.com/mitchellh/go-testing-interface".T (missing Cleanup method)
10:52:37 avi/data_source_avi_applicationpersistenceprofile_test.go:13:15: cannot use t (type *"testing".T) as type "github.com/mitchellh/go-testing-interface".T in argument to resource.Test:
10:52:37 	*"testing".T does not implement "github.com/mitchellh/go-testing-interface".T (missing Cleanup method)
10:52:37 avi/data_source_avi_applicationprofile_test.go:13:15: cannot use t (type *"testing".T) as type "github.com/mitchellh/go-testing-interface".T in argument to resource.Test:
10:52:37 	*"testing".T does not implement "github.com/mitchellh/go-testing-interface".T (missing Cleanup method)
10:52:37 avi/data_source_avi_autoscalelaunchconfig_test.go:13:15: cannot use t (type *"testing".T) as type "github.com/mitchellh/go-testing-interface".T in argument to resource.Test:
10:52:37 	*"testing".T does not implement "github.com/mitchellh/go-testing-interface".T (missing Cleanup method)
10:52:37 avi/data_source_avi_certificatemanagementprofile_test.go:10:15: cannot use t (type *"testing".T) as type "github.com/mitchellh/go-testing-interface".T in argument to resource.Test:
10:52:37 	*"testing".T does not implement "github.com/mitchellh/go-testing-interface".T (missing Cleanup method)
10:52:37 avi/data_source_avi_cloud_test.go:13:15: cannot use t (type *"testing".T) as type "github.com/mitchellh/go-testing-interface".T in argument to resource.Test:
10:52:37 	*"testing".T does not implement "github.com/mitchellh/go-testing-interface".T (missing Cleanup method)
10:52:37 avi/data_source_avi_cloud_test.go:13:15: too many errors
10:52:37 make: *** [testacc] Error 2
```

@yograjshisode , might need to generate the cleanup funcs or TF Tests to satisfy interface or hardcode the version that satisfies our requirements in go.mod

File: go.mod (hunk 1 above)
Comment (yograjshisode) [reply go.mod:22]: We can consume these now once tests are okay

**Pull request closed without merging**